src/librustc/traits/error_reporting.rs

```diff
@@ -347,6 +347,53 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         }
     }
 
+    fn describe_generator(&self, body_id: hir::BodyId) -> Option<&'static str> {
+        self.tcx.hir().body(body_id).generator_kind.map(|gen_kind| {
+            match gen_kind {
+                hir::GeneratorKind::Gen => { "a generator" },
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Block) => { "an async block" },
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Fn) => { "an async function" },
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Closure)
+                    => { "an async closure" },
```

Comment:
Use the braces to put `"an async closure"` on its own line. When using braces the trailing comma is not necessary after the block.